src/Stack/Types/Build.hs

```diff
@@ -617,7 +618,9 @@ configureOptsNoDir :: EnvConfig
 configureOptsNoDir econfig bco deps isLocal package = concat
     [ depOptions
     , ["--enable-library-profiling" | boptsLibProfile bopts || boptsExeProfile bopts]
-    , ["--enable-executable-profiling" | boptsExeProfile bopts && isLocal]
+    -- Cabal < 1.21.1 does not support --enable-profiling, use --enable-executable-profiling instead
+    , let profFlag = "--enable-" <> concat ["executable-" | newerCabal] <> "profiling"
```

Comment:
I think this is the wrong way around now – we want to use `--enable-profiling` for the more recent versions of `Cabal`.